### PR TITLE
[IOTDB-3955] Improved node information metrics to push more detailed node information, online and offline total and survival information of each node

### PR DESCRIPTION
**Pull request by 23931017wu**
#### The survival status of the current node is represented by 1:online,0:unkonwn.
data_node{name="EndPoint(127.0.0.1:6671)",status="1:online,0:unkonwn",} 1.0
config_node{name="EndPoint(0.0.0.0:22277)",status="1:online,0:unkonwn",} 1.0
#### The total number of registed nodes is represented by registed.
data_node{name="total",status="registed",} 1.0
config_node{name="total",status="registed",} 1.0
#### The total number of online nodes is represented by online.
data_node{name="total",status="online",} 2.0
config_node{name="total",status="online",} 2.0
#### The total number of unknown nodes is represented by unknown.
data_node{name="total",status="unknown",} 1.0
config_node{name="total",status="unknown",} 1.0
### Finally think of a sample
image
image
image

#### Start the 3C3D first 
image
#### Displayed on the Grafana dashboard(Just to show)
image
#### Stop 1 ConfigNode and 1 DataNode
image
#### Displayed on the Grafana dashboard(Just to show)
image

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java [hunk 1]
```diff
@@ -411,6 +420,257 @@ public List<TDataNodeConfiguration> getOnlineDataNodes(int dataNodeId) {
...
+                      + ")";
+      MetricsService.getInstance()
+              .getMetricManager()
+              .getOrCreateGauge(
+                      Metric.CONFIG_NODE.toString(),
+                      MetricLevel.IMPORTANT,
+                      Tag.NAME.toString(),
+                      name,
+                      Tag.STATUS.toString(),
+                      "1:online,0:unkonwn")
```
Comment (mychaow) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:466]: status should be online or unknown? no need to public 1 or 0

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java [hunk 2]
```diff
@@ -112,7 +112,9 @@ public void addMetrics() {
               registeredConfigNodes,
               o -> getRegisteredConfigNodeCount(),
               Tag.NAME.toString(),
-              "online");
+              "total",
+              Tag.STATUS.toString(),
+              "registed");
```
Comment (mychaow) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:117]: registered

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java [hunk 3]
```diff
@@ -411,6 +420,257 @@ public List<TDataNodeConfiguration> getOnlineDataNodes(int dataNodeId) {
...
+    }
+    return allConfigNodes.size();
+  }
+
+  public int getUnknownDataNodesNum() {
+    List<TDataNodeConfiguration> allDataNodes = getUnknownDataNodes(-1);
+    if(allDataNodes == null){
+      return 0;
+    }
+    for (TDataNodeConfiguration dataNodeInfo : allDataNodes) {
```
Comment (mychaow) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:530]: I think the confignode/datanode ip metric is too many. 
And I do not think the online time is important.  
In current implement, if I delete some datanode, but the metric exists.

**Review by mychaow — APPROVED** [review]
LGTM

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java [hunk 4]
```diff
@@ -244,6 +249,8 @@ public void start() {
 
   /** Stop the heartbeat service and the load balancing service */
   public void stop() {
+    LOGGER.info("Remove loadManager metrics");
```
Comment (SpriCoder) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:252]: This log is useless.

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java [hunk 5]
```diff
@@ -420,6 +428,140 @@ public List<TDataNodeConfiguration> getOnlineDataNodes(int dataNodeId) {
...
+          .getOrCreateGauge(
+              Metric.DATA_NODE.toString(),
+              MetricLevel.CORE,
+              Tag.NAME.toString(),
+              "total",
+              Tag.STATUS.toString(),
+              "unknown")
+          .set(getUnknownDataNodesNum());
+    }
+  }
```
Comment (SpriCoder) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:528]: I think the value of tag can be the name of NodeStatus enums to remove hard code.

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java [hunk 6]
```diff
@@ -112,7 +112,9 @@ public void addMetrics() {
               registeredConfigNodes,
               o -> getRegisteredConfigNodeCount(),
               Tag.NAME.toString(),
-              "online");
+              "total",
+              Tag.STATUS.toString(),
+              "registered");
```
Comment (SpriCoder) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:117]: I think the value of tag can be the name of NodeStatus enums to remove hard code.

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java [hunk 7]
```diff
@@ -121,7 +123,9 @@ public void addMetrics() {
               registeredDataNodes,
               Map::size,
               Tag.NAME.toString(),
-              "online");
+              "total",
+              Tag.STATUS.toString(),
+              "registered");
```
Comment (SpriCoder) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/NodeInfo.java:128]: I think the value of tag can be the name of NodeStatus enums to remove hard code.

**Review by SpriCoder — CHANGES_REQUESTED** [review]
PTAL~
Besides, I have two other problems:
1. Please update metric doc at the same time: docs/UserGuide/Maintenance-Tools/Metric-Tool.md and docs/zh/UserGuide/Maintenance-Tools/Metric-Tool.md
2. Maybe you can update the dashboard at the same time.

**Review by neuyilan — APPROVED** [review]
LGTM

**Pull request merged**